content/rationale.md

```diff
@@ -0,0 +1,85 @@
+title: Why complate?
+
+Why complate?
+=============
+
+## Abstraction
+
+When we are developing our frontend components, we can use complate markup
+abstractions ("macros") as an abstraction layer for our component. Consider
+the following macro:
+
+```jsx
+function Navbar (_props, ...children) {
+  return <nav>
+    {children}
+  </nav>
+}
+```
+
+We can then use our macro in our application and it will be replaced with the
+markup we have defined.
+
+```jsx
+<Navbar />
+```
+
+This is _extremely powerful_ and solves a major pain point when writing
+maintainable frontend code. Requirements change constantly. When a new request
+comes, or the corporate design has changed yet again, I can change the markup
+of my component _without breaking any part of my application that uses the
+component_.
```

Comment:
That seems like a potentially misleading promise, as various other parts might rely on that markup (CSS, JavaScript, tests). Perhaps we could argue with encapsulation instead: Change the definition and all instances are updated automatically? I might be overly cautious though.

More generally, I'd argue with less specific scenarios - in particular, external/corporate requirements being the defining factor kinda offends my anti-authoritarian/-capitalist proclivities; in my experience, markup adjustments are often intrinsically motivated. (Obviously that, too, is an FNDism, so YMMV.)